litestar/utils/typing.py

```diff
@@ -133,6 +135,12 @@
 wrapper_type_set = {Annotated, Required, NotRequired}
 """Types that always contain a wrapped type annotation as their first arg."""
 
+# `strict` was only introduced in 3.10
+if sys.version_info >= (3, 10):
+    _zip = partial(zip, strict=True)
```

Comment:
I think we shouldn't do this, since this means that the behaviour will be different for 3.10 and above.

Reply:
That makes sense. I'll remove the strict check, because the other option is to implement zip ourselves with strict check for versions lower than 3.10, which I don't think is a good idea.